src/borg/manifest.py

```diff
@@ -37,6 +37,8 @@ class MandatoryFeatureUnsupported(Error):
 def filter_archives_by_date(archives, older=None, newer=None, oldest=None, newest=None):
     def get_first_and_last_archive_ts(archives_list):
         timestamps = [x.ts for x in archives_list]
+        if not timestamps:
+            return None, None
```

Comment:
how is line 49 working with the None values returned here?

Reply:
@jetchirag did you see my feedback?

Reply:
@ThomasWaldmann I can't remember exactly right now. I'll check it thoroughly and get back. Perhaps I made a mistake.

Reply:
I had a look, guess it could be fixed like this:
- add some tests that fail with the current code (see bug report)
- there is one call to `filter_archives_by_date` and before doing that, it checks for `len(archives) > 0` - remove that check.
- in `filter_archives_by_date`, check twice for the archives list being empty:
  - at the very beginning
  - after the archives list has been modified and before calling get_first_and_last_archive_ts
  - if archives list is empty, just `return archives`
- tests should work now

Reply:
@ThomasWaldmann I just pushed your suggested code. I've appended to the test of original feature PR. Do we need more test?

Reply:
yes, see my other comment. by the main issues are resolved.